substrate/frame/nomination-pools/test-delegate-stake/src/mock.rs

```diff
@@ -42,29 +42,11 @@ pub(crate) const POOL1_REWARD: AccountId = 20397359637244482196168876781421u128;
 
 #[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
 impl frame_system::Config for Runtime {
-	type BaseCallFilter = frame_support::traits::Everything;
-	type BlockWeights = ();
-	type BlockLength = ();
-	type DbWeight = ();
-	type RuntimeOrigin = RuntimeOrigin;
 	type Nonce = Nonce;
-	type RuntimeCall = RuntimeCall;
-	type Hash = sp_core::H256;
-	type Hashing = sp_runtime::traits::BlakeTwo256;
 	type AccountId = AccountId;
 	type Lookup = IdentityLookup<Self::AccountId>;
```

Comment:
Is there a reason why the `TestDefaultConfig` is explicitly u64 instead of `IdentityLookup<Self::AccountId>` - I don't know the internals of `derive_impl`, but would that not mean that we could always leave it out even when we want to override the `AccountId` type? Maybe one for @gupnik

Reply:
I recall there was a technical issue -- a default for `type A` cannot rely on `type B`  in the same trait.

Reply:
Good one @seadanda. I think it should be possible now @kianenigma. Let me look into this and raise a PR if it works.